### DISCUSSION TITLE
Argument buffers max sampler count question

### DISCUSSION THREAD

**Discussion by Benau** [KhronosGroup/MoltenVK#1637] [Q&A]
Hello, I wonder if that conuts against unique samplers because:

```
layout(binding = 0) uniform sampler2D tex[4096];
```
works which different images are used together with only single sampler (combined image sampler)

where getMaxSamplerCount() returns 2048 for me, and I assume it should only work when tex[2048] (or not?)

or we should not determine base on VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_DESCRIPTOR_INDEXING_PROPERTIES because they are arbitrary?
```
pDescIdxProps->maxPerStageDescriptorUpdateAfterBindSamplers			= isTier2 ? maxSampCnt : _properties.limits.maxPerStageDescriptorSamplers;
pDescIdxProps->maxPerStageDescriptorUpdateAfterBindUniformBuffers	= isTier2 ? 500000 : _properties.limits.maxPerStageDescriptorUniformBuffers;
pDescIdxProps->maxPerStageDescriptorUpdateAfterBindStorageBuffers	= isTier2 ? 500000 : _properties.limits.maxPerStageDescriptorStorageBuffers;
pDescIdxProps->maxPerStageDescriptorUpdateAfterBindSampledImages	= isTier2 ? 500000 : _properties.limits.maxPerStageDescriptorSampledImages;
pDescIdxProps->maxPerStageDescriptorUpdateAfterBindStorageImages	= isTier2 ? 500000 : _properties.limits.maxPerStageDescriptorStorageImages;
```

Thank you

**Comment by jaremieromer** [comment]
According to the Vulkan spec - yes:

maxDescriptorSetSamplers is the maximum number of samplers that can be included in a pipeline layout.
Descriptors with a type of VK_DESCRIPTOR_TYPE_SAMPLER or VK_DESCRIPTOR_TYPE_COMBINED_IMAGE_SAMPLER count against this limit. Only descriptors in descriptor set layouts created without the VK_DESCRIPTOR_SET_LAYOUT_CREATE_UPDATE_AFTER_BIND_POOL_BIT bit set count against this limit.
 
 I have essentially the same question with some additional questions here, so I don't want to make a separate post as this post captures the essence of my question:
 
 Since it seems that:

1.  MacOS/MoltenVK can only support 80 unique samplers, https://vulkan.gpuinfo.org/displaycoreproperty.php?platform=macos&name=maxDescriptorSetUpdateAfterBindSamplers&core=1.2
2. CombinedImageSamplers count against that requirement, even if only one sampler is shared amongst all sampled images
3. SPIR-V does not support splitting out the sampler and texture2D if a texture2D bindless array is used

Does that mean we are limited to only 80 textures bound to a pipeline at any given time?
